### PR TITLE
Change from custom package manager to vcpkg.

### PR DESCRIPTION
**Pull request by Dwordcito**
## Description

This pr aims to add the vcpkg package manager to this project.

### CODE REVIEW

**Review by sebasfalcone — COMMENTED** [review]
I am not able to compile the project, I am having some issues with the dependencies. I will keep investigating this issue

**Comment by Dwordcito** [comment]
> I am not able to compile the project, I am having some issues with the dependencies. I will keep investigating this issue

README updated.

File: .github/workflows/http-request-ci.yml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        uses: lukka/run-vcpkg@v11
+        with:
+          vcpkgDirectory: '${{github.workspace}}/vcpkg'
+          vcpkgGitCommitId: 'a42af01b72c28a8e1d7b48107b33e4f286a55ef6'
+          vcpkgJsonGlob: '${{github.workspace}}/vcpkg.json'
+      - name: Configure CMake
+        run: cmake --preset=release
+
+      - name: Build
+        run: cmake --build build -j$(nproc)
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply .github/workflows/http-request-ci.yml:98]: I just activated the test compilation, only in debug, the idea is to test that the project compiles and then see if we can go further with the tests.

**Review by sebasfalcone — APPROVED** [review]
- Code 🟢 
- Tests 🟢 
```
./urlrequest_testtool -u https://httpbin.org/delete -t delete
{
  "args": {}, 
  "data": "", 
  "files": {}, 
  "form": {}, 
  "headers": {
    "Accept": "application/json", 
    "Accept-Charset": "utf-8", 
    "Authorization": "Basic Og==", 
    "Content-Type": "application/json", 
    "Host": "httpbin.org", 
    "X-Amzn-Trace-Id": "Root=1-66cdc91d-29a6309752e8952e03656603"
  }, 
  "json": null, 
  "origin": "190.19.48.135", 
  "url": "https://httpbin.org/delete"
}

./urlrequest_testtool -u https://httpbin.org/post -t post -p out
{
  "args": {}, 
  "data": "", 
  "files": {}, 
  "form": {}, 
  "headers": {
    "Accept": "application/json", 
    "Accept-Charset": "utf-8", 
    "Authorization": "Basic Og==", 
    "Content-Length": "0", 
    "Content-Type": "application/json", 
    "Host": "httpbin.org", 
    "X-Amzn-Trace-Id": "Root=1-66cdc910-4d7040804ff52cba02d557b0"
  }, 
  "json": null, 
  "origin": "190.19.48.135", 
  "url": "https://httpbin.org/post"
}

./urlrequest_testtool -u https://httpbin.org/get -t download -o out
cat out 
{
  "args": {}, 
  "headers": {
    "Accept": "application/json", 
    "Accept-Charset": "utf-8", 
    "Authorization": "Basic Og==", 
    "Content-Type": "application/json", 
    "Host": "httpbin.org", 
    "X-Amzn-Trace-Id": "Root=1-66cdc8f8-5654c2cc7c2a9a3a35ec641b"
  }, 
  "origin": "190.19.48.135", 
  "url": "https://httpbin.org/get"
}
```

**Pull request merged**